lib/pub_cache.dart

```diff
@@ -90,15 +105,18 @@ class Application {
   final Directory _dir;
 
   List<PackageRef> _packageRefs;
-  Version _version;
 
   Application._(this._cache, this._dir);
 
   String get name => path.basename(_dir.path);
 
-  Version get version {
-    if (_packageRefs == null) _parsePubspecLock();
-    return _version;
+  Version get version => getDefiningPackageRef().version;
```

Comment:
It looks like `getDefiningPackageRef()` might return `null` in which case this'll fall down.  Maybe a check?